pycyphal/transport/udp/_ip/_v4.py

```diff
@@ -139,13 +152,16 @@ def make_input_socket(self, data_specifier: pycyphal.transport.DataSpecifier) ->
         return s
 
     def make_sniffer(self, handler: typing.Callable[[LinkLayerCapture], None]) -> SnifferIPv4:
-        return SnifferIPv4(self._local, handler)
+        return SnifferIPv4(self._subnet_id, handler)
 
 
 class SnifferIPv4(Sniffer):
-    def __init__(self, local_ip_address: IPv4Address, handler: typing.Callable[[LinkLayerCapture], None]) -> None:
-        netmask_width = IPV4LENGTH - IP_ADDRESS_NODE_ID_MASK.bit_length()
-        subnet = ip_network(f"{local_ip_address}/{netmask_width}", strict=False)
+    def __init__(self, handler: typing.Callable[[LinkLayerCapture], None]) -> None:
+        netmask_width = IPV4LENGTH - DESTINATION_NODE_ID_MASK.bit_length() - 1
```

Comment:
This usage of `DESTINATION_NODE_ID_MASK` is incorrect because it neglects subject addresses and attempts to filter traffic based on the IP address of the origin, which is no longer meaningful within the Cyphal/UDP stack.

Due to this, the BPF expression below needs to be modified such that it filters the traffic based on the destination address, not the origin; the local IP address of the current node is therefore no longer needed at all: `udp and dst net 239.0.0.0/15`. I chose 15 here to also include the five bits reserved for the domain-ID.

I did not have the opportunity to test this, though. You can do that using Wireshark; enter the above expression before initiating the capture on the network interface (keep in mind that the capture filter uses the BPF syntax while the display filter uses some Wireshark-specific filter that is not related to BPF although it occasionally looks deceptively similar).